ClassNotFoundException from KryoHelper.readObject when running ContestEstimator for seata-core-0.5.0

**Description**

Running Contest Estimator log for seata-core-0.5.0:
Symbolic execution FAILED for Symbolic execution FAILED for `io.seata.core.rpc.netty.RmRpcClient.sendResponse`
There are 12 of ClassNotFoundExceptions for MockitoException from Concrete Executor.

**To Reproduce**

1. Run the 'utbot' project in IntelliJ Idea 2022.2.4
2. Open `utbot-junit-contest/src/main/kotlin/org/utbot/contest/ContestEstimator.kt`
3. Change `javaHome` property to **_full local path to JDK 8_**, like "C:\\\\Program Files\\\\Amazon Corretto\\\\jdk1.8.0_332"
4. Change `timeLimit` to 60
5. In `fun main` change: `projectFilter = listOf("seata-core-0.5.0")`
6. Run `fun main` - make sure to set JRE in Edit Configurations to JDK 8.

**Expected behavior**

No exceptions are supposed to be thrown from Concrete Executor.

**Actual behavior**

There are 12 of ClassNotFoundExceptions for MockitoException from Concrete Executor - for different classes.

**Visual proofs (screenshots, logs, images)**

The following test is generated:

```java
///region Test suites for executable io.seata.core.rpc.netty.RmRpcClient.sendResponse
    
    ///region
    
    @Test
    public void testSendResponse1() throws Exception  {
        org.mockito.MockedStatic mockedStatic = null;
        try {
            Class rmRpcClientClazz = Class.forName("io.seata.core.rpc.netty.RmRpcClient");
            Logger prevLOGGER = ((Logger) getStaticFieldValue(rmRpcClientClazz, "LOGGER"));
            try {
                setStaticField(rmRpcClientClazz, "LOGGER", null);
                mockedStatic = mockStatic(org.slf4j.LoggerFactory.class);
                (mockedStatic.when(() -> org.slf4j.LoggerFactory.getLogger(any(Class.class)))).thenReturn(((Logger) null));
                RmRpcClient rmRpcClient = ((RmRpcClient) createInstance("io.seata.core.rpc.netty.RmRpcClient"));
                Object object = new Object();
                
                /* This test fails because method [io.seata.core.rpc.netty.RmRpcClient.sendResponse] produces [java.lang.NullPointerException]
                    io.seata.core.rpc.netty.RmRpcClient.sendResponse(RmRpcClient.java:404) */
                rmRpcClient.sendResponse(0L, ((String) null), object);
            } finally {
                setStaticField(RmRpcClient.class, "LOGGER", prevLOGGER);
            }
        } finally {
            mockedStatic.close();
        }
    }
    ///endregion
    
    ///endregion
```

Log extract:

~~~java
12:17:52.085 | INFO  |  -- Remaining time budget: 44038 ms, #remaining_methods: 17, budget for method: 2590 ms, solver timeout: 1000 ms, budget for symbolic execution: 1390 ms, budget for concrete execution: 1200 ms,  -- 
12:17:52.174 | DEBUG | --new testCase collected, to generate: testIo.seata.core.rpc.netty.RmRpcClient.sendResponse1
12:17:53.484 | INFO  | |> Starting concrete execution for remaining state: io.seata.core.rpc.netty.RmRpcClient.sendResponse
12:17:53.501 | ERROR | Symbolic execution FAILED
org.utbot.instrumentation.util.InstrumentedProcessError: Error in the instrumented process |> org.utbot.instrumentation.util.ReadingFromKryoException: Reading from Kryo exception |> com.esotericsoftware.kryo.kryo5.KryoException: Error during Java deserialization.
Serialization trace:
exception (org.utbot.framework.plugin.api.UtImplicitlyThrownException)
result (org.utbot.framework.concrete.UtConcreteExecutionResult)
	at com.esotericsoftware.kryo.kryo5.serializers.JavaSerializer.read(JavaSerializer.java:68)
	at com.esotericsoftware.kryo.kryo5.Kryo.readObject(Kryo.java:796)
	at com.esotericsoftware.kryo.kryo5.serializers.ReflectField.read(ReflectField.java:123)
	at com.esotericsoftware.kryo.kryo5.serializers.FieldSerializer.read(FieldSerializer.java:129)
	at com.esotericsoftware.kryo.kryo5.Kryo.readObject(Kryo.java:796)
	at com.esotericsoftware.kryo.kryo5.serializers.ReflectField.read(ReflectField.java:123)
	at com.esotericsoftware.kryo.kryo5.serializers.FieldSerializer.read(FieldSerializer.java:129)
	at com.esotericsoftware.kryo.kryo5.Kryo.readClassAndObject(Kryo.java:877)
	at org.utbot.instrumentation.util.KryoHelper.readObject(KryoHelper.kt:94)
	at org.utbot.instrumentation.ConcreteExecutor$executeAsync$2.invokeSuspend(ConcreteExecutor.kt:246)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: com.esotericsoftware.kryo.kryo5.KryoException: Class not found: org.mockito.exceptions.base.MockitoException
	at com.esotericsoftware.kryo.kryo5.serializers.JavaSerializer$ObjectInputStreamWithKryoClassLoader.resolveClass(JavaSerializer.java:88)
	at java.base/java.io.ObjectInputStream.readNonProxyDesc(ObjectInputStream.java:1997)
	at java.base/java.io.ObjectInputStream.readClassDesc(ObjectInputStream.java:1864)
	at java.base/java.io.ObjectInputStream.readOrdinaryObject(ObjectInputStream.java:2195)
	at java.base/java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1681)
	at java.base/java.io.ObjectInputStream.readObject(ObjectInputStream.java:493)
	at java.base/java.io.ObjectInputStream.readObject(ObjectInputStream.java:451)
	at com.esotericsoftware.kryo.kryo5.serializers.JavaSerializer.read(JavaSerializer.java:66)
	... 17 more
Caused by: java.lang.ClassNotFoundException: org.mockito.exceptions.base.MockitoException
	at java.base/java.net.URLClassLoader.findClass(URLClassLoader.java:476)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:589)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
	at java.base/java.lang.Class.forName0(Native Method)
	at java.base/java.lang.Class.forName(Class.java:398)
	at com.esotericsoftware.kryo.kryo5.serializers.JavaSerializer$ObjectInputStreamWithKryoClassLoader.resolveClass(JavaSerializer.java:86)
	... 24 more

	at org.utbot.instrumentation.util.KryoHelper.readObject(KryoHelper.kt:96)
	at org.utbot.instrumentation.ConcreteExecutor$executeAsync$2.invokeSuspend(ConcreteExecutor.kt:246)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: com.esotericsoftware.kryo.kryo5.KryoException: Error during Java deserialization.
Serialization trace:
exception (org.utbot.framework.plugin.api.UtImplicitlyThrownException)
result (org.utbot.framework.concrete.UtConcreteExecutionResult)
	at com.esotericsoftware.kryo.kryo5.serializers.JavaSerializer.read(JavaSerializer.java:68)
	at com.esotericsoftware.kryo.kryo5.Kryo.readObject(Kryo.java:796)
	at com.esotericsoftware.kryo.kryo5.serializers.ReflectField.read(ReflectField.java:123)
	at com.esotericsoftware.kryo.kryo5.serializers.FieldSerializer.read(FieldSerializer.java:129)
	at com.esotericsoftware.kryo.kryo5.Kryo.readObject(Kryo.java:796)
	at com.esotericsoftware.kryo.kryo5.serializers.ReflectField.read(ReflectField.java:123)
	at com.esotericsoftware.kryo.kryo5.serializers.FieldSerializer.read(FieldSerializer.java:129)
	at com.esotericsoftware.kryo.kryo5.Kryo.readClassAndObject(Kryo.java:877)
	at org.utbot.instrumentation.util.KryoHelper.readObject(KryoHelper.kt:94)
	... 9 more
Caused by: com.esotericsoftware.kryo.kryo5.KryoException: Class not found: org.mockito.exceptions.base.MockitoException
	at com.esotericsoftware.kryo.kryo5.serializers.JavaSerializer$ObjectInputStreamWithKryoClassLoader.resolveClass(JavaSerializer.java:88)
	at java.base/java.io.ObjectInputStream.readNonProxyDesc(ObjectInputStream.java:1997)
	at java.base/java.io.ObjectInputStream.readClassDesc(ObjectInputStream.java:1864)
	at java.base/java.io.ObjectInputStream.readOrdinaryObject(ObjectInputStream.java:2195)
	at java.base/java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1681)
	at java.base/java.io.ObjectInputStream.readObject(ObjectInputStream.java:493)
	at java.base/java.io.ObjectInputStream.readObject(ObjectInputStream.java:451)
	at com.esotericsoftware.kryo.kryo5.serializers.JavaSerializer.read(JavaSerializer.java:66)
	... 17 more
Caused by: java.lang.ClassNotFoundException: org.mockito.exceptions.base.MockitoException
	at java.base/java.net.URLClassLoader.findClass(URLClassLoader.java:476)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:589)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
	at java.base/java.lang.Class.forName0(Native Method)
	at java.base/java.lang.Class.forName(Class.java:398)
	at com.esotericsoftware.kryo.kryo5.serializers.JavaSerializer$ObjectInputStreamWithKryoClassLoader.resolveClass(JavaSerializer.java:86)
	... 24 more

	at org.utbot.instrumentation.ConcreteExecutor.withProcess(ConcreteExecutor.kt:227) ~[main/:?]
	at org.utbot.instrumentation.ConcreteExecutor$withProcess$1.invokeSuspend(ConcreteExecutor.kt) ~[main/:?]
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33) [kotlin-stdlib-1.7.20.jar:1.7.20-release-201(1.7.20)]
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106) [kotlinx-coroutines-core-jvm-1.6.3.jar:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:829) [?:?]
Caused by: org.utbot.instrumentation.util.ReadingFromKryoException: Reading from Kryo exception |> com.esotericsoftware.kryo.kryo5.KryoException: Error during Java deserialization.
Serialization trace:
exception (org.utbot.framework.plugin.api.UtImplicitlyThrownException)
result (org.utbot.framework.concrete.UtConcreteExecutionResult)
	at com.esotericsoftware.kryo.kryo5.serializers.JavaSerializer.read(JavaSerializer.java:68)
	at com.esotericsoftware.kryo.kryo5.Kryo.readObject(Kryo.java:796)
	at com.esotericsoftware.kryo.kryo5.serializers.ReflectField.read(ReflectField.java:123)
	at com.esotericsoftware.kryo.kryo5.serializers.FieldSerializer.read(FieldSerializer.java:129)
	at com.esotericsoftware.kryo.kryo5.Kryo.readObject(Kryo.java:796)
	at com.esotericsoftware.kryo.kryo5.serializers.ReflectField.read(ReflectField.java:123)
	at com.esotericsoftware.kryo.kryo5.serializers.FieldSerializer.read(FieldSerializer.java:129)
	at com.esotericsoftware.kryo.kryo5.Kryo.readClassAndObject(Kryo.java:877)
	at org.utbot.instrumentation.util.KryoHelper.readObject(KryoHelper.kt:94)
	at org.utbot.instrumentation.ConcreteExecutor$executeAsync$2.invokeSuspend(ConcreteExecutor.kt:246)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: com.esotericsoftware.kryo.kryo5.KryoException: Class not found: org.mockito.exceptions.base.MockitoException
	at com.esotericsoftware.kryo.kryo5.serializers.JavaSerializer$ObjectInputStreamWithKryoClassLoader.resolveClass(JavaSerializer.java:88)
	at java.base/java.io.ObjectInputStream.readNonProxyDesc(ObjectInputStream.java:1997)
	at java.base/java.io.ObjectInputStream.readClassDesc(ObjectInputStream.java:1864)
	at java.base/java.io.ObjectInputStream.readOrdinaryObject(ObjectInputStream.java:2195)
	at java.base/java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1681)
	at java.base/java.io.ObjectInputStream.readObject(ObjectInputStream.java:493)
	at java.base/java.io.ObjectInputStream.readObject(ObjectInputStream.java:451)
	at com.esotericsoftware.kryo.kryo5.serializers.JavaSerializer.read(JavaSerializer.java:66)
	... 17 more
Caused by: java.lang.ClassNotFoundException: org.mockito.exceptions.base.MockitoException
	at java.base/java.net.URLClassLoader.findClass(URLClassLoader.java:476)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:589)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
	at java.base/java.lang.Class.forName0(Native Method)
	at java.base/java.lang.Class.forName(Class.java:398)
	at com.esotericsoftware.kryo.kryo5.serializers.JavaSerializer$ObjectInputStreamWithKryoClassLoader.resolveClass(JavaSerializer.java:86)
	... 24 more

	at org.utbot.instrumentation.util.KryoHelper.readObject(KryoHelper.kt:96) ~[main/:?]
	at org.utbot.instrumentation.ConcreteExecutor$executeAsync$2.invokeSuspend(ConcreteExecutor.kt:246) ~[main/:?]
	... 8 more
Caused by: com.esotericsoftware.kryo.kryo5.KryoException: Error during Java deserialization.
Serialization trace:
exception (org.utbot.framework.plugin.api.UtImplicitlyThrownException)
result (org.utbot.framework.concrete.UtConcreteExecutionResult)
	at com.esotericsoftware.kryo.kryo5.serializers.JavaSerializer.read(JavaSerializer.java:68) ~[kryo5-5.3.0.jar:?]
	at com.esotericsoftware.kryo.kryo5.Kryo.readObject(Kryo.java:796) ~[kryo5-5.3.0.jar:?]
	at com.esotericsoftware.kryo.kryo5.serializers.ReflectField.read(ReflectField.java:123) ~[kryo5-5.3.0.jar:?]
	at com.esotericsoftware.kryo.kryo5.serializers.FieldSerializer.read(FieldSerializer.java:129) ~[kryo5-5.3.0.jar:?]
	at com.esotericsoftware.kryo.kryo5.Kryo.readObject(Kryo.java:796) ~[kryo5-5.3.0.jar:?]
	at com.esotericsoftware.kryo.kryo5.serializers.ReflectField.read(ReflectField.java:123) ~[kryo5-5.3.0.jar:?]
	at com.esotericsoftware.kryo.kryo5.serializers.FieldSerializer.read(FieldSerializer.java:129) ~[kryo5-5.3.0.jar:?]
	at com.esotericsoftware.kryo.kryo5.Kryo.readClassAndObject(Kryo.java:877) ~[kryo5-5.3.0.jar:?]
	at org.utbot.instrumentation.util.KryoHelper.readObject(KryoHelper.kt:94) ~[main/:?]
	at org.utbot.instrumentation.ConcreteExecutor$executeAsync$2.invokeSuspend(ConcreteExecutor.kt:246) ~[main/:?]
	... 8 more
Caused by: com.esotericsoftware.kryo.kryo5.KryoException: Class not found: org.mockito.exceptions.base.MockitoException
	at com.esotericsoftware.kryo.kryo5.serializers.JavaSerializer$ObjectInputStreamWithKryoClassLoader.resolveClass(JavaSerializer.java:88) ~[kryo5-5.3.0.jar:?]
	at java.io.ObjectInputStream.readNonProxyDesc(ObjectInputStream.java:1997) ~[?:?]
	at java.io.ObjectInputStream.readClassDesc(ObjectInputStream.java:1864) ~[?:?]
	at java.io.ObjectInputStream.readOrdinaryObject(ObjectInputStream.java:2195) ~[?:?]
	at java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1681) ~[?:?]
	at java.io.ObjectInputStream.readObject(ObjectInputStream.java:493) ~[?:?]
	at java.io.ObjectInputStream.readObject(ObjectInputStream.java:451) ~[?:?]
	at com.esotericsoftware.kryo.kryo5.serializers.JavaSerializer.read(JavaSerializer.java:66) ~[kryo5-5.3.0.jar:?]
	at com.esotericsoftware.kryo.kryo5.Kryo.readObject(Kryo.java:796) ~[kryo5-5.3.0.jar:?]
	at com.esotericsoftware.kryo.kryo5.serializers.ReflectField.read(ReflectField.java:123) ~[kryo5-5.3.0.jar:?]
	at com.esotericsoftware.kryo.kryo5.serializers.FieldSerializer.read(FieldSerializer.java:129) ~[kryo5-5.3.0.jar:?]
	at com.esotericsoftware.kryo.kryo5.Kryo.readObject(Kryo.java:796) ~[kryo5-5.3.0.jar:?]
	at com.esotericsoftware.kryo.kryo5.serializers.ReflectField.read(ReflectField.java:123) ~[kryo5-5.3.0.jar:?]
	at com.esotericsoftware.kryo.kryo5.serializers.FieldSerializer.read(FieldSerializer.java:129) ~[kryo5-5.3.0.jar:?]
	at com.esotericsoftware.kryo.kryo5.Kryo.readClassAndObject(Kryo.java:877) ~[kryo5-5.3.0.jar:?]
	at org.utbot.instrumentation.util.KryoHelper.readObject(KryoHelper.kt:94) ~[main/:?]
	at org.utbot.instrumentation.ConcreteExecutor$executeAsync$2.invokeSuspend(ConcreteExecutor.kt:246) ~[main/:?]
	... 8 more
Caused by: java.lang.ClassNotFoundException: org.mockito.exceptions.base.MockitoException
	at java.net.URLClassLoader.findClass(URLClassLoader.java:476) ~[?:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:589) ~[?:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:522) ~[?:?]
	at java.lang.Class.forName0(Native Method) ~[?:?]
	at java.lang.Class.forName(Class.java:398) ~[?:?]
	at com.esotericsoftware.kryo.kryo5.serializers.JavaSerializer$ObjectInputStreamWithKryoClassLoader.resolveClass(JavaSerializer.java:86) ~[kryo5-5.3.0.jar:?]
	at java.io.ObjectInputStream.readNonProxyDesc(ObjectInputStream.java:1997) ~[?:?]
	at java.io.ObjectInputStream.readClassDesc(ObjectInputStream.java:1864) ~[?:?]
	at java.io.ObjectInputStream.readOrdinaryObject(ObjectInputStream.java:2195) ~[?:?]
	at java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1681) ~[?:?]
	at java.io.ObjectInputStream.readObject(ObjectInputStream.java:493) ~[?:?]
	at java.io.ObjectInputStream.readObject(ObjectInputStream.java:451) ~[?:?]
	at com.esotericsoftware.kryo.kryo5.serializers.JavaSerializer.read(JavaSerializer.java:66) ~[kryo5-5.3.0.jar:?]
	at com.esotericsoftware.kryo.kryo5.Kryo.readObject(Kryo.java:796) ~[kryo5-5.3.0.jar:?]
	at com.esotericsoftware.kryo.kryo5.serializers.ReflectField.read(ReflectField.java:123) ~[kryo5-5.3.0.jar:?]
	at com.esotericsoftware.kryo.kryo5.serializers.FieldSerializer.read(FieldSerializer.java:129) ~[kryo5-5.3.0.jar:?]
	at com.esotericsoftware.kryo.kryo5.Kryo.readObject(Kryo.java:796) ~[kryo5-5.3.0.jar:?]
	at com.esotericsoftware.kryo.kryo5.serializers.ReflectField.read(ReflectField.java:123) ~[kryo5-5.3.0.jar:?]
	at com.esotericsoftware.kryo.kryo5.serializers.FieldSerializer.read(FieldSerializer.java:129) ~[kryo5-5.3.0.jar:?]
	at com.esotericsoftware.kryo.kryo5.Kryo.readClassAndObject(Kryo.java:877) ~[kryo5-5.3.0.jar:?]
	at org.utbot.instrumentation.util.KryoHelper.readObject(KryoHelper.kt:94) ~[main/:?]
	at org.utbot.instrumentation.ConcreteExecutor$executeAsync$2.invokeSuspend(ConcreteExecutor.kt:246) ~[main/:?]
	... 8 more
12:17:53.507 | ERROR | Symbolic execution FAILED ... <<stack trace duplicated>>
12:17:53.512 | ERROR | Symbolic execution FAILED ... <<stack trace duplicated>>
12:17:53.518 | ERROR | Symbolic execution FAILED ... <<stack trace duplicated>>
12:17:53.524 | ERROR | Symbolic execution FAILED ... <<stack trace duplicated>>
12:17:53.529 | ERROR | Symbolic execution FAILED ... <<stack trace duplicated>>
12:17:53.535 | ERROR | Symbolic execution FAILED ... <<stack trace duplicated>>
12:17:53.541 | ERROR | Symbolic execution FAILED ... <<stack trace duplicated>>
12:17:53.661 | INFO  | Finished (in 1,576 sec): method io.seata.core.rpc.netty.RmRpcClient.sendResponse 
<StatsForMethod> :
	#generatedTC=1
	FAILED 8 time(s) with 1 different exception(s)"
~~~

**Environment**

IntelliJ IDEA 2022.2.4
JDK 8

--- a/buildSrc/src/main/java/SettingsTemplateHelper.java
+++ b/buildSrc/src/main/java/SettingsTemplateHelper.java
@@ -47,7 +47,7 @@ public class SettingsTemplateHelper {
                 if (s.startsWith("enum class ")) {//Enum class declaration
                     enums.add(new EnumInfo(s.substring(11, s.length() - 2)));
                 } else if (s.matches("[A-Z_]+,?") && !enums.isEmpty()) {//Enum value
-                    var enumValue = s.substring(0, s.length());
+                    String enumValue = s.substring(0, s.length());
                     if (enumValue.endsWith(",")) enumValue = enumValue.substring(0, enumValue.length() - 1);
                     enums.get(enums.size() - 1).docMap.put(enumValue, new ArrayList<>(docAcc));
                 } else if (s.startsWith("const val ")) {//Constand value to be substitute later if need
@@ -66,9 +66,9 @@ public class SettingsTemplateHelper {
                     s = acc.toString();
                     acc.delete(0, acc.length());
                     if (s.startsWith("var") || s.startsWith("val")) {
-                        var i = s.indexOf(" by ", 3);
+                        int i = s.indexOf(" by ", 3);
                         if (i > 0) {
-                            var key = s.substring(3, i).trim();
+                            String key = s.substring(3, i).trim();
                             if (key.contains(":")) {
                                 key = key.substring(0, key.lastIndexOf(':'));
                             }
@@ -91,7 +91,7 @@ public class SettingsTemplateHelper {
                             i = s.indexOf('(', i);
                             if (i > 0) {
                                 s = s.substring(i + 1);
-                                var defaultValue = s.substring(0, s.indexOf(')')).trim();
+                                String defaultValue = s.substring(0, s.indexOf(')')).trim();
                                 if (defaultValue.contains(",")) defaultValue = defaultValue.substring(0, defaultValue.indexOf(','));
                                 defaultValue = dictionary.getOrDefault(defaultValue, defaultValue);
                                 if (defaultValue.matches("[\\d_]+L")) {

--- a/utbot-instrumentation/src/main/kotlin/org/utbot/instrumentation/util/JavaSerializerWrapper.kt
+++ b/utbot-instrumentation/src/main/kotlin/org/utbot/instrumentation/util/JavaSerializerWrapper.kt
@@ -1,0 +1,42 @@
+package org.utbot.instrumentation.util
+
+import com.esotericsoftware.kryo.kryo5.Kryo
+import com.esotericsoftware.kryo.kryo5.KryoException
+import com.esotericsoftware.kryo.kryo5.io.Input
+import com.esotericsoftware.kryo.kryo5.serializers.JavaSerializer
+import java.io.InputStream
+import java.io.ObjectInputStream
+import java.io.ObjectStreamClass
+
+/**
+ * This ad-hoc solution for ClassNotFoundException
+ */
+class JavaSerializerWrapper : JavaSerializer() {
+    override fun read(kryo: Kryo, input: Input?, type: Class<*>?): Any? {
+        return try {
+            val graphContext = kryo.graphContext
+            var objectStream = graphContext.get<Any>(this) as ObjectInputStream?
+            if (objectStream == null) {
+                objectStream = WrappingObjectInputStream(input, kryo)
+                graphContext.put(this, objectStream)
+            }
+            objectStream.readObject()
+        } catch (ex: java.lang.Exception) {
+            throw KryoException("Error during Java deserialization.", ex)
+        }
+    }
+}
+
+class WrappingObjectInputStream(iss : InputStream?, private val kryo: Kryo) : ObjectInputStream(iss) {
+    override fun resolveClass(type: ObjectStreamClass): Class<*>? {
+        return try {
+            Class.forName(type.name, false, kryo.classLoader)
+        } catch (ex: ClassNotFoundException) {
+            try {
+                return Kryo::class.java.classLoader.loadClass(type.name)
+            } catch (e: ClassNotFoundException) {
+                throw KryoException("Class not found: " + type.name, ex)
+            }
+        }
+    }
+}

--- a/utbot-instrumentation/src/main/kotlin/org/utbot/instrumentation/util/JavaSerializerWrapper.kt
+++ b/utbot-instrumentation/src/main/kotlin/org/utbot/instrumentation/util/JavaSerializerWrapper.kt
@@ -35,7 +35,11 @@ class WrappingObjectInputStream(iss : InputStream?, private val kryo: Kryo) : Ob
             try {
                 return Kryo::class.java.classLoader.loadClass(type.name)
             } catch (e: ClassNotFoundException) {
-                throw KryoException("Class not found: " + type.name, ex)
+                try {
+                    return super.resolveClass(type);
+                } catch (e: ClassNotFoundException) {
+                    throw KryoException("Class not found: " + type.name, e)
+                }
             }
         }
     }

--- a/utbot-instrumentation/src/main/kotlin/org/utbot/instrumentation/util/KryoHelper.kt
+++ b/utbot-instrumentation/src/main/kotlin/org/utbot/instrumentation/util/KryoHelper.kt
@@ -8,12 +8,9 @@ import com.esotericsoftware.kryo.kryo5.io.Output
 import com.esotericsoftware.kryo.kryo5.objenesis.instantiator.ObjectInstantiator
 import com.esotericsoftware.kryo.kryo5.objenesis.strategy.InstantiatorStrategy
 import com.esotericsoftware.kryo.kryo5.objenesis.strategy.StdInstantiatorStrategy
-import com.esotericsoftware.kryo.kryo5.serializers.JavaSerializer
 import com.esotericsoftware.kryo.kryo5.util.DefaultInstantiatorStrategy
 import com.jetbrains.rd.util.lifetime.Lifetime
 import com.jetbrains.rd.util.lifetime.throwIfNotAlive
-import org.utbot.api.exception.UtMockAssumptionViolatedException
-import org.utbot.framework.plugin.api.TimeoutException
 import java.io.ByteArrayOutputStream
 
 /**
@@ -123,47 +120,14 @@ internal class TunedKryo : Kryo() {
         }
 
         this.setOptimizedGenerics(false)
-        register(TimeoutException::class.java, TimeoutExceptionSerializer())
-        register(UtMockAssumptionViolatedException::class.java, UtMockAssumptionViolatedExceptionSerializer())
 
         // TODO: JIRA:1492
-        addDefaultSerializer(java.lang.Throwable::class.java, JavaSerializer())
+        addDefaultSerializer(java.lang.Throwable::class.java, JavaSerializerWrapper())
 
         val factory = object : SerializerFactory.FieldSerializerFactory() {}
         factory.config.ignoreSyntheticFields = true
         factory.config.serializeTransient = false
         factory.config.fieldsCanBeNull = true
         this.setDefaultSerializer(factory)
-    }
-
-    /**
-     * Specific serializer for [TimeoutException] - [JavaSerializer] is not applicable
-     * because [TimeoutException] is not in class loader.
-     *
-     * This serializer is very simple - it just writes [TimeoutException.message]
-     * because we do not need other components.
-     */
-    private class TimeoutExceptionSerializer : Serializer<TimeoutException>() {
-        override fun write(kryo: Kryo, output: Output, value: TimeoutException) {
-            output.writeString(value.message)
-        }
-
-        override fun read(kryo: Kryo?, input: Input, type: Class<out TimeoutException>?): TimeoutException =
-            TimeoutException(input.readString())
-    }
-
-    /**
-     * Specific serializer for [UtMockAssumptionViolatedException] - [JavaSerializer] is not applicable
-     * because [UtMockAssumptionViolatedException] is not in class loader.
-     */
-    private class UtMockAssumptionViolatedExceptionSerializer : Serializer<UtMockAssumptionViolatedException>() {
-        override fun write(kryo: Kryo, output: Output, value: UtMockAssumptionViolatedException) {
-            output.writeString(value.message)
-        }
-
-        override fun read(kryo: Kryo?, input: Input, type: Class<out UtMockAssumptionViolatedException>?): UtMockAssumptionViolatedException {
-            input.readString() // shift the reading position
-            return UtMockAssumptionViolatedException()
-        }
     }
 }

